src/main/java/step2/StringCalculator.java

```diff
@@ -0,0 +1,90 @@
+package step2;
+
+import java.util.Arrays;
+
+public class StringCalculator {
+
+    private static final String[] DEFAULT_DELIMITERS = {",", ":"};
+    private static final String[] CUSTOM_DELIMITERS = {"//", "\n"};
+
+    public int calculate(String string) {
+        if (hasNotValue(string)) {
+            return 0;
+        }
+
+        string = convertToSameDelimiter(string);
+        int[] numbers = validate(string.split(DEFAULT_DELIMITERS[1]));
+
+        return sum(numbers);
+    }
+
+    private boolean hasNotValue(String string) {
+        return string == null || string.isBlank() || string.isEmpty();
+    }
+
+    private String convertToSameDelimiter(String string) {
+        if (isDefaultDelimiter(string)) {
+            string = string.replace(DEFAULT_DELIMITERS[0], DEFAULT_DELIMITERS[1]);
+        }
+
+        if (isCustomDelimiter(string)) {
+            string = convertToDefaultDelimiter(string);
+        }
+
+        return string;
+    }
```

Comment:
매개변수로 들어온 변수를 조건에 따라 중간에 값을 변경하고 그 값을 리턴하는 방식은 안티패턴일 수 있어요.
전달된 값을 메서드가 끝날 때까지 따라가야하는 가독이 좋지 못한 코드가 만들어질 가능성이 높아요! 🤔
지금은 간단한 로직이기에 크게 와닿진 않겠지만, 복잡한 로직일 수록 가독성이 떨어질 확률이 높습니다 🫠

이런 선언 방식을 사용하지 않고 코드를 작성해보시길 권장드립니다 🙂

ex)
```
String a = "a";
if(any) {
  a = "b"
}
return a;
```

와 같은 코드는
```
if(any) {
  return "b"
}
return "a";
```
로 리펙토링될 수 있습니다.

```suggestion
    private String convertToSameDelimiter(String string) {
        if (isCustomDelimiter(string)) {
            return convertToDefaultDelimiter(string);
        }
        if (!isDefaultDelimiter(string)) {
            // 예외
        }

        return string.replace(DEFAULT_DELIMITERS[0], DEFAULT_DELIMITERS[1]);
    }
```

이렇게 될 수 있을것 같아요! 😄

Reply:
좋은 의견 감사합니다! 몰랐던 부분을 알게 돼서 기분이 너무 좋네요 :smile:

민수님 의견을 바탕으로 리팩토링을 진행하며 '구분자가 없는 문자열'에 대한 케이스를 조금 더 명시적으로 표현 할 필요성을 느껴 제안하신 코드와는 다소 다르게 리팩토링 됐습니다!